### PR TITLE
15 new elements + 1 new modifier!

### PR DESCRIPTION
**Pull request by nayakrujul**
<s>Tests coming soon™</s> All done :)

### CODE REVIEW

File: shared/src/vyxal/ListHelpers.scala [hunk 1]
```diff
@@ -670,6 +670,11 @@ object ListHelpers:
   def gradeUp(iterable: VAny)(using Context): VList =
     VList.from(makeIterable(iterable).zipWithIndex.sortBy(_._1).map(_._2))
 
+  def gradeDown(iterable: VAny)(using Context): VList =
+    VList.from(
+      makeIterable(iterable).zipWithIndex.sortBy(_._1).reverse.map(_._2)
```
Comment (Lyxal) [inline shared/src/vyxal/ListHelpers.scala:675]: Iirc there's some cases where this doesn't actually return the list graded down.

Same reason I had to change the version 2 implementation from this method. It didn't match up with what apl and jelly produced.

**Review by Lyxal — CHANGES_REQUESTED** [review]

File: shared/src/vyxal/ListHelpers.scala (hunk 1 above)
Comment (nayakrujul) [reply shared/src/vyxal/ListHelpers.scala:675]: Oh yeah, when the list isn't unique.

Comment (nayakrujul) [reply shared/src/vyxal/ListHelpers.scala:675]: Can you or @ysthakur fix it? I'm not sure how to.

Comment (nayakrujul) [reply shared/src/vyxal/ListHelpers.scala:675]: @Lyxal I'm pretty sure this *is* the same method as v2. I tested it with non-unique lists, and it gives the same results as v2's `⇩`.

**Comment by nayakrujul** [comment]
@Lyxal fixed! I added a test case for you to verify, which I checked against [the v2 output](https://vyxal.pythonanywhere.com/?v=2#WyIiLCIiLCLih6kiLCIiLCIxLCAyLCAxLCAyLCAxIl0=).

**Comment by Lyxal** [comment]
I'll have to find that challenge where I realised the implementation was broken to add that as a test case.

**Comment by Lyxal** [comment]
https://lyxal.pythonanywhere.com?flags=P&code=%E2%82%8C%E2%87%A7%E2%87%A9W&inputs=%5B0%2C1%2C1%2C0%2C0%2C1%2C0%2C1%2C1%5D&header=&footer=

**Review by Lyxal — APPROVED** [review]
Pre-approving awaiting tests

File: shared/src/vyxal/Modifiers.scala [hunk 2]
```diff
@@ -408,6 +408,29 @@ object Modifiers:
             AST.Command("#|map-prefixes"),
           )
       },
+    "ᶳ" ->
+      Modifier(
+        "Sort By",
+        """|Sort By Element
+           |ᶳf: Sort top of stack based on results of f""".stripMargin,
+        List("sort-by"),
```
Comment (Lyxal) [inline shared/src/vyxal/Modifiers.scala:416]:
```suggestion
        List("sort-by:"),
```

**Review by Lyxal — CHANGES_REQUESTED** [review]

**Review by Lyxal — CHANGES_REQUESTED** [review]
```
[1,2,3,4,5,6,7,8] sort-by: {negate}
```

is broken. That's a skill issue.

File: shared/src/vyxal/ListHelpers.scala [hunk 3]
```diff
@@ -670,6 +670,13 @@ object ListHelpers:
   def gradeUp(iterable: VAny)(using Context): VList =
     VList.from(makeIterable(iterable).zipWithIndex.sortBy(_._1).map(_._2))
 
+  def gradeDown(iterable: VAny)(using Context): VList =
+    VList.from(
+      makeIterable(iterable).zipWithIndex
+        .sorted(Ordering.by((a: (VAny, Int)) => a._1).reverse)
```
[the remark this answers is not in the document]
Comment (nayakrujul) [reply shared/src/vyxal/ListHelpers.scala:676]: Negate won't work on strings, will it?

**Comment by nayakrujul** [comment]
@ysthakur yes, `a.real.sqrt` works.

**Comment by nayakrujul** [comment]
@ysthakur I don't think that works: Wolfram Alpha says [`sqrt(1+4i) != 1+2i`](https://www.wolframalpha.com/input?i2d=true&i=Sqrt%5B1%2B4i%5D)

**Comment by nayakrujul** [comment]
@ysthakur done!

File: shared/src/vyxal/Elements.scala [hunk 4]
```diff
@@ -2001,6 +2148,36 @@ object Elements:
...
+    ) { ctx ?=>
+      ctx.pop() match
+        case f: VFun =>
+          val arg = ListHelpers.makeIterable(ctx.pop())
+          ctx.push(arg.maxBy(v => f(v)))
+        case _ => throw IllegalArgumentException(
+            "Maximum By: First argument should be a function"
+          )
+    },
+    addDirect(
```
[the remark this answers is not in the document]
Comment (nayakrujul) [reply shared/src/vyxal/Elements.scala:2166]: That was @Lyxal; we should probably ask him before doing anything else.

File: shared/src/vyxal/Elements.scala [hunk 5]
```diff
@@ -1906,6 +2041,18 @@ object Elements:
       case a: VNum => NumberHelpers.primeFactors(a).distinct
       case a: String => StringHelpers.caseof(a)
     },
+    addPart(
+      Dyad,
+      "Φ",
+      "Umm I have no clue what to call this @Lyxal @ysthakur please help",
```
[the remark this answers is not in the document]
Comment (nayakrujul) [reply shared/src/vyxal/Elements.scala:2047]: Yeah that's better lol

**Review by Steffan153 — APPROVED** [review]

**Pull request merged**